IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php

```diff
@@ -117,6 +132,24 @@ function post()
                         $category = trim($category);
                         if ($category) {
                             $content .= " #$category";
+                $categories    = $this->getInput('category');
+		$location = $this->getInput('location');
+
+                if ($type == 'entry') {
+                    $type = 'article';
+                    if (!empty($_FILES['photo'])) {
+                        $type = 'photo';
+                    } else {
+                        $photo_url = $this->getInput('photo');
+                        if ($photo_url) {
+                            $type      = 'photo';
+                            $success   = $this->uploadFromUrl($photo_url);
+                            if (!$success) {
+                            	\Idno\Core\Idno::site()->triggerEvent('indiepub/post/failure', ['page' => $this]);
+                                $this->setResponse(500);
+                                echo "Failed uploading photo from $photo_url";
+                                exit;
+                            }
```

Comment:
this block seems to have migrated down from above (starting around 81 in the old file)? maybe git made a mistake merging?